kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -206,3 +209,96 @@ neqKey = "KEQUAL.neq"
 
 iteKey :: IsString s => s
 iteKey = "KEQUAL.ite"
+
+{- | The @KEQUAL.eq@ hooked symbol applied to @term@-type arguments.
+ -}
+data KEqual term =
+    KEqual
+        { symbol :: !Symbol
+        , operand1, operand2 :: !term
+        }
+
+{- | Match the @KEQUAL.eq@ hooked symbol.
+ -}
+matchKEqual :: TermLike variable -> Maybe (KEqual (TermLike variable))
+matchKEqual (App_ symbol [operand1, operand2]) = do
+    hook2 <- (getHook . symbolHook) symbol
+    Monad.guard (hook2 == eqKey)
+    return KEqual { symbol, operand1, operand2 }
+matchKEqual _ = Nothing
+
+termKEqualsTrue
+    :: forall variable unifier
+    .  InternalVariable variable
+    => MonadUnify unifier
+    => TermSimplifier variable unifier
+    -> TermLike variable
+    -> TermLike variable
+    -> MaybeT unifier (Pattern variable)
+termKEqualsTrue unifyChildren a b =
+    worker a b <|> worker b a
+  where
+    unifyChildren' term1 term2 =
+        unificationPredicate term1 term2
+        <|> return (makeEqualsPredicate_ term1 term2)
+    unificationPredicate term1 term2 =
+        Condition.toPredicate
+        . Pattern.withoutTerm
+        <$> unifyChildren term1 term2
+    atLeastOneSymbolic term1 term2 =
+        case (Builtin.toKey term1, Builtin.toKey term2) of
+           (Nothing, _) -> True
+           (_, Nothing) -> True
+           _ -> False
+    worker termLike1 termLike2
+      | Just KEqual { operand1, operand2 } <- matchKEqual termLike1
+      , isFunctionPattern termLike1
+      , Just value2 <- Bool.matchBool termLike2
+      , value2
+      , atLeastOneSymbolic operand1 operand2
+      = lift $ do
+        condition <- unifyChildren' operand1 operand2
+        pure Conditional
+            { term = mkTop (termLikeSort termLike1)
```

Comment:
The term should be `termLike2`; what you have here is only correct when we have `(_ ==K _) = true`, but not when we have `(_ ==K _) ∧ true`. The second form is likely to appear in `owise` rules.